tests/Definitions/DefinitionGeneratorTest.php

```diff
@@ -646,4 +642,93 @@ private function assertHasDefinitions(array $definitions)
             });
         }
     }
-}
\ No newline at end of file
+
+    public function testGetAllRelations(): void
+    {
+        $route = $this->newRouteByName('customers.store');
+
+        $generator = new DefinitionGenerator($route, []);
+
+        $relations = $generator->getAllRelations(new Customer());
+        $relations = array_column($relations, 'related_model');
+
+        $relation_names = array_map('class_basename', $relations);
+
+        $this->assertEquals([
+            'Order',
+            'Company',
+            'Address'
+        ], $relation_names);
+    }
+
+    public function testOnlyRelationMethodsInvoked(): void
+    {
+        $mockCustomer = $this->createMock(Customer::class);
+        $route = $this->newRouteByName('customers.store');
+
+        $generator = new DefinitionGenerator($route, []);
+
+        $mockCustomer->expects($this->never())
+            ->method('someOtherMethod');
+        $mockCustomer->expects($this->once())
+            ->method('orders');
+        $mockCustomer->expects($this->once())
+            ->method('company');
+        $mockCustomer->expects($this->once())
+            ->method('address');
+
+        $generator->getAllRelations($mockCustomer);
+    }
+}
+
+class Customer extends Model
+{
+    protected function someProtectedMethod()
+    {
+        return [];
+    }
+
+    public function someOtherMethod()
+    {
+        return 'blah';
+    }
+
+    public function orders(): HasMany
+    {
+        return $this->hasMany('Mtrajano\LaravelSwagger\Tests\Definitions\Order');
+    }
+
+    public function company(): BelongsTo
+    {
+        return $this->belongsTo('Mtrajano\LaravelSwagger\Tests\Definitions\Company');
+    }
+
+    public function address(): HasOne
+    {
+        return $this->hasOne('Mtrajano\LaravelSwagger\Tests\Definitions\Address');
+    }
+}
+
+class Order extends Model
+{
+    public function customer()
+    {
+        return $this->belongsTo('Mtrajano\LaravelSwagger\Tests\Definitions\Customer');
+    }
+}
+
+class Company extends Model
+{
+    public function customers()
+    {
+        return $this->hasMany('Mtrajano\LaravelSwagger\Tests\Definitions\Customer');
+    }
+}
+
+class Address extends Model
+{
+    public function customer()
+    {
+        return $this->belongsTo('Mtrajano\LaravelSwagger\Tests\Definitions\Customer');
+    }
```

Comment:
going to clean up these tests and move them to the stubs namespace during a later refactor after the merge